bolt-design-system/bolt
# Button | HTML | Add missing 'disabled' attribute to button and trigger

## Jira

http://vjira2:8080/browse/BDS-1563

## Summary
Add `disabled` attribute to `bolt-button` and `bolt-trigger` inner elements when prop is set.

## Details

Bolt button and trigger components do not currently add the `disabled` attribute to the inner semantic `<button>` element even though they receive the proper visual style. Without the `disabled` attribute `<button>` is still focusable according to the Accessibility Object Model.

## How to test

- Given I am running the `fix/add-button-disabled-attribute` branch locally
- When I go to the [Button style variation page](http://localhost:3000/pattern-lab/patterns/02-components-button-15-button-style-variations/02-components-button-15-button-style-variations.html) and I inspect each "disabled" button example
- Then I see the `disabled` attribute on each component's inner `<button>` tag
- When I go to the [Trigger disabled variation page](http://localhost:3000/pattern-lab/patterns/02-components-trigger-30-trigger-disabled-variations/02-components-trigger-30-trigger-disabled-variations.html) and I inspect each "disabled" trigger example
- Then I see the `disabled` attribute on the first example's inner element and `aria-disabled="true"`  on the second, and the second does not have an `href` attribute
- When I paste the following JS into the console and I inspect the four web components added to the page:
```
var el = document.createElement('div');
el.innerHTML = `<div><bolt-button disabled>Basic button (disabled)</bolt-button></div>
                <div><bolt-button disabled url="http://pega.com">Basic button link (disabled)</bolt-button></div>
                <div><bolt-trigger disabled>Basic trigger (disabled)</bolt-trigger></div>
                <div><bolt-trigger url="http://pega.com" disabled>Basic trigger link (unchanged by 'disabled' prop)</bolt-trigger></div>`;
document.body.appendChild(el);
```
- Then I see the `disabled` attribute on the inner `<button>` tags and `aria-disabled="true"` on the inner `<a>` tags, and `<a>` tags do not have `href` attributes
